### PR TITLE
Helm chart: add CI GitHub Actions workflow

### PR DESCRIPTION
**Pull request by desaintmartin**
CD will come in a few days.

What it does:
- Linting chart, detecting changes
- Validating all generated resources from all test cases (`ci/*-values.yaml`) files against kubernetes schema
- Deploying in kind and testing validity for all test cases

I've removed the deploy script as it does not fit the un-coupling of chart version from dashboard version.

### CODE REVIEW

File: aio/deploy/helm-chart/kubernetes-dashboard/README.md [hunk 1]
```diff
@@ -131,7 +131,11 @@ helm install kubernetes-dashboard/kubernetes-dashboard --name my-release -f valu
 
 > **Tip**: You can use the default [values.yaml](values.yaml), which is used by default, as reference
 
-## Using the dashboard with 'kubectl proxy'
+## Access
+
+For information about how to access, please read the [kubernetes-dashboard manual](https://kubernetes.io/docs/tasks/access-application-cluster/web-ui-dashboard/)
```
Comment (floreks) [inline aio/deploy/helm-chart/kubernetes-dashboard/README.md:136]: Please point to our README instead. Kubernetes docs are usually not up-to-date.

File: .helm-chart-testing.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+chart-dirs:
```
Comment (floreks) [inline .helm-chart-testing.yaml:1]: Can we move this file to i.e. helm chart dir?

**Comment by desaintmartin** [comment]
Nice, the CI will now fail if chart version has not been bumped but if Chart has been modified.

**Comment by desaintmartin** [comment]
Change requests have been addressed.

**Comment by desaintmartin** [comment]
Rebased.

File: .github/workflows/ci-helm-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  push:
+    branches: [ master ]
+    paths:
+    - 'aio/deploy/helm-chart/**'
+  pull_request:
+    paths:
+    - 'aio/deploy/helm-chart/**'
+
+jobs:
+  helm-lint-test:
```
Comment (floreks) [inline .github/workflows/ci-helm-workflow.yml:13]: nit: Can you name the job? Similar to how it's done for other jobs: https://github.com/kubernetes/dashboard/blob/665b4d367d61dc429d3765982ced3c4aab08daa7/.github/workflows/ci-workflow.yml#L9-L12

Comment (desaintmartin) [reply .github/workflows/ci-helm-workflow.yml:13]: Done!

**Pull request merged**